pennylane/tape/tape.py

```diff
@@ -416,7 +416,7 @@ def _process_queue(self):
             elif isinstance(obj, qml.operation.Operation) and not info.get("owner", False):
                 # operation objects with no owners
 
-                if self._measurements:
+                if any(m for m in self._measurements if m.return_type != qml.operation.MidMeasure):
```

Comment:
Good catch! This can be reverted because mid-circuit measurements won't end up in `tape._measurements`.

Yep, I think `mid_measurements` could be nice to have. One thing is that there are many examples (including the new transforms paper) that assume the
1. `tape._ops`
2. `tape._measurements`

order when iterating over operations. If we made major changes, such examples have to be updated. Hence this PR doesn't bear that weight yet.